### PR TITLE
Timestamp wrapper

### PR DESCRIPTION
**Pull request by itkovian**
Remove common code from scripts and have it here.

### CODE REVIEW

File: test/timestamp.py [hunk 1]
```diff
@@ -126,4 +129,11 @@ def test_convert_timestamp(self):
...
         self.assertEqual(convert_timestamp(date), (datet, date))
 
+    @mock.patch('vsc.utils.timestamp.read_timestamp')
+    def test_retrieve_timestamp(self, mock_read_timestamp):
+        """Test for the filestamp retrieval."""
+
+        mock_read_timestamp.return_value = "203412130101"
 
+        self.assertEqual("20140102", retrieve_timestamp_with_default("f", start_timestamp="20140102")[0])
+        self.assertEqual(2049584460, retrieve_timestamp_with_default("f", default_timestamp="20140102")[0])
```
Comment (stdweird) [inline test/timestamp.py]: add one more test for returned current_timetistamp
```
self.assertEqual(utc, retrieve_timestamp_with_default("f", default_timestamp="20140102")[1].tzinfo)
```
it's

File: lib/vsc/utils/timestamp.py [hunk 2]
```diff
@@ -131,3 +131,22 @@ def write_timestamp(filename, timestamp):
...
+    timestamp = start_timestamp
+    if not start_timestamp:
+        timestamp = convert_to_unix_timestamp(read_timestamp(filename))
+
+    timestamp = timestamp or default_timestamp
+
+    logging.info("Using timestamp %s", start_timestamp)
+
+    current_time = datetime.datetime.now(tz=utc) + datetime.timedelta(seconds=delta)
+    return (timestamp, current_time)
```
Comment (stdweird) [inline lib/vsc/utils/timestamp.py]: i would also add a function to do
```
            (_, ldap_timestamp) = convert_timestamp(start_time)
            write_timestamp(SYNC_TIMESTAMP_FILENAME, ldap_timestamp)
```
or support new option to write_timestamp to do the conversion

File: test/timestamp.py [hunk 3]
```diff
@@ -126,4 +129,11 @@ def test_convert_timestamp(self):
         self.assertEqual(convert_timestamp("20171029010500Z"), (datet, date))
         self.assertEqual(convert_timestamp(date), (datet, date))
 
+    @mock.patch('vsc.utils.timestamp.read_timestamp')
+    def test_retrieve_timestamp(self, mock_read_timestamp):
```
Comment (wpoely86) [inline test/timestamp.py]: what happens when the file doesn't exist or contains gibberish?

File: lib/vsc/utils/timestamp.py [hunk 4]
```diff
@@ -131,3 +131,22 @@ def write_timestamp(filename, timestamp):
...
+    Return a tuple consisting of the following values:
+    - the timestamp from the given file if the start_timestamp is not provided 
+      and fall back on the default if needed.
+    - the current time based on the given delta, defaulting to 10s.
+    """
+    timestamp = start_timestamp
+    if not start_timestamp:
+        timestamp = convert_to_unix_timestamp(read_timestamp(filename))
+
+    timestamp = timestamp or default_timestamp
```
Comment (wpoely86) [inline lib/vsc/utils/timestamp.py]: so what will timestamp be when the file doesn't exist?

File: lib/vsc/utils/timestamp.py [hunk 5]
```diff
@@ -131,3 +131,22 @@ def write_timestamp(filename, timestamp):
...
+
+    logging.info("Using timestamp %s", start_timestamp)
+
+    current_time = datetime.datetime.now(tz=utc) + datetime.timedelta(seconds=delta)
```
[the remark this answers is not in the document]
Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: Depends. We can give the delta as a value < 0. Or we could turn this into a minute. But I think a delta is always added, even when one wants to reduce the number.

[the remark this answers is not in the document]
Comment (stdweird) [reply lib/vsc/utils/timestamp.py]: then the default should be -10, not +10

File: test/timestamp.py (hunk 3 above)
Comment (itkovian) [reply test/timestamp.py]: That's the responsibility of another function. Jens decided that if the file contains garbage, the `read_timestamp` returns `None` and is addressed in the conversion function, where it defaults to `today()`.

So, I think I'd better change that behaviour and convert the default if None is "read".

Comment (stdweird) [reply test/timestamp.py]: i think we should fix that in an other PR, as it will affect other repos too

Comment (wpoely86) [reply test/timestamp.py]: can you add a test then that checks that if the file contains gibberish, the default timestamp is used instead?

Comment (itkovian) [reply test/timestamp.py]: yes.

File: lib/vsc/utils/timestamp.py [hunk 6]
```diff
@@ -131,3 +131,22 @@ def write_timestamp(filename, timestamp):
...
     cache.update('timestamp', timestamp_, 0)
     cache.close()
+
+
+def retrieve_timestamp_with_default(filename, start_timestamp=None, default_timestamp="2014010100000Z", delta=10):
+    """
+    Return a tuple consisting of the following values:
+    - the timestamp from the given file if the start_timestamp is not provided 
+      and fall back on the default if needed.
+    - the current time based on the given delta, defaulting to 10s.
```
[the remark this answers is not in the document]
Comment (stdweird) [reply lib/vsc/utils/timestamp.py]: "based on" -> "+"

File: lib/vsc/utils/timestamp.py (hunk 2 above)
Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: The question then really becomes if we still need this, no? If we'd store unix epoch everywhere we'd be fine, no?

Comment (stdweird) [reply lib/vsc/utils/timestamp.py]: again, different PR as that would affect multiple repos
i don't mind stporing it n multiple formats, but the value should be correct ;)

the 2nd value is the one that is used and is human readable, so i'd say non-trivial change

Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: Yes, that would be my main concern.

Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: OK, I'll keep the above for another PR then.

File: lib/vsc/utils/timestamp.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: Fixed.

**Pull request merged**